### ISSUE TITLE
Builder reactivity broken when using previews

### ISSUE THREAD

**Issue by martin-ro** [laravel-filament/filament#13971]
### Package

filament/filament

### Package Version

v3.2.99

### Laravel Version

v11.20.0

### PHP Version

PHP 8.2

### Problem description

When using the `blockPreviews` for `Builder` the reactivity in the modal form doesn't work once the block is saved.

https://github.com/user-attachments/assets/0fb1a824-7025-4f7e-937e-9d1523ea96d1

### Expected behavior

Same as without `blockPreviews`

### Steps to reproduce

See https://github.com/martin-ro/block-builder (UserResource)

or

add this to a from:

```php
Forms\Components\Builder::make('blocks')
                    ->blockPreviews()
                    ->schema([
                        Forms\Components\Builder\Block::make('test')
                            ->preview('components.test')
                            ->schema([
                                Forms\Components\Checkbox::make('is_company')
                                    ->live(),

                                Forms\Components\TextInput::make('company_name')
                                    ->hidden(fn(Forms\Get $get): bool => !$get('is_company'))
                            ])
                    ])
```

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/martin-ro/block-builder

**Comment by martin-ro** [comment]
Another observation I just made is that wrapping components in for example a group or fieldset reactivity works inside it. So this is only broken for "top level" components.

**Comment by borjajimnz** [comment]
@danharrin this seems to be already fixed, at least in the latest version [[v3.2.136](https://github.com/filamentphp/filament/releases/tag/v3.2.136)]

https://youtu.be/SNWtRdyG2Sg  

Additionally, the given repository does not exist.
regards

**Comment by danharrin** [comment]
@martin-ro can you confirm please?

**Comment by martin-ro** [comment]
> [@martin-ro](https://github.com/martin-ro) can you confirm please?

Yes @danharrin, the issue persists. The yt video @borjajimnz posted never saved the model so it appears it is working. 

Steps to reproduce:
1. Add `Test` block
2. Check checkbox
3. Enter name
4. Save block
5. Save record

Reactivity works so far

6. Uncheck checkbox
7. Save block
8. Save record

Now reactivity is broken, you can not get the `TextInput` to be shown anymore.

I pushed another repo here: https://github.com/martin-ro/filament-reactivity/

On the `UserResource` there is a commented second example that works because the fields in the block are wrapped in a group.

**Comment by borjajimnz** [comment]
Yes, @martin-ro, I can reproduce this now. I'll give it a shot. Thanks!
